platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,322 @@
+      if (fileFromConfigAdmin != null) {
+        // The felix file prop changed, which is not allowed (3)
+        String msg =
+            format(
+                "Felix filename has been illegally changed from [%s] to [%s]",
+                fileFromCache, fileFromConfigAdmin);
+        throw new IllegalStateException(msg);
+      }
+      if (fileFromCache.exists()) {
+        // The felix file prop was removed, which we can revert if the file exists (3)
+        LOGGER.info("Felix filename has been illegally removed, reverting to [{}]", fileFromCache);
+        context.setProperty(FELIX_FILENAME, fileFromCache.getAbsolutePath());
+        writeIfNecessary(
+            appropriatePid, fileFromCache, context.getSanitizedProperties(), cachedConfigData);
+        return;
+      }
+    }
+
+    // Routine property updates for tracked files - write to disk if necessary (4)
+    writeIfNecessary(
+        appropriatePid, fileFromConfigAdmin, context.getSanitizedProperties(), cachedConfigData);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Remove config files from etc when a delete occurs but only if felix is tracking the file.
+   *
+   * @param pid the pid pointing to the configuration to delete.
+   */
+  @Override
+  public void handleDelete(String pid) throws IOException {
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Object>)
+              () -> {
+                doHandleDelete(pid);
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
+      // Only caught if the action threw a CHECKED exception, thus it MUST be an IOException.
+      throw (IOException) e.getCause();
+    }
+  }
+
+  private void doHandleDelete(String pid) throws IOException {
+    CachedConfigData cachedConfigData = pidDataMap.remove(pid);
+    File fileFromCache = (cachedConfigData != null) ? cachedConfigData.getFelixFile() : null;
+    if (fileFromCache != null && fileFromCache.exists()) {
+      LOGGER.debug("Deleting file because config was deleted for pid [{}]", pid);
+      try {
+        Files.delete(fileFromCache.toPath());
+        SecurityLogger.audit("Removing a deleted config [{}]", fileFromCache.toPath());
+      } catch (IOException e) {
+        LOGGER.debug(
+            format("Problem deleting config file [%s]: ", fileFromCache.getAbsolutePath()), e);
+        // Synchronous with config admin, so we can report the failure to the UI this way
+        throw e;
+      }
+    }
+  }
+
+  private void writeIfNecessary(
+      String appropriatePid,
+      File dest,
+      Dictionary<String, Object> configState,
+      CachedConfigData cachedData)
+      throws IOException {
+    if (dest != null && dest.exists()) {
+      processPasswords(appropriatePid, configState, this::encryptValue);
+      if (!cachedData.equalProps(configState)) {
+        try (OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(dest))) {
```

Comment:
❗️ One problem here is that your audit for failure below doesn't catch the IOException that could occur on close from the try-with-resource.

Reply:
Verified in a test bed application that exceptions thrown by the `close()` method **will** be caught in this blocks present state. An embedded try does not appear to be necessary.

Reply:
Gist for the test bed: https://gist.github.com/Lambeaux/ae75875f6ac89d9f94830705841dd068